src/System.Windows.Extensions/tests/System/Drawing/Printing/MarginsConverterTests.cs

```diff
@@ -0,0 +1,172 @@
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                Assert.True(mc.CanConvertFrom(context, typeof(string)));
+                Assert.False(mc.CanConvertFrom(context, typeof(Guid)));
+                Assert.False(mc.CanConvertFrom(context, typeof(object)));
+                Assert.False(mc.CanConvertFrom(context, typeof(int)));
+            }
+        }
+
+        [Fact]
+        public void CanConvertTo()
+        {
+            MarginsConverter mc = new MarginsConverter();
+
+            // try once with then once without context
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                Assert.True(mc.CanConvertTo(context, typeof(string)));
+                Assert.False(mc.CanConvertTo(context, typeof(Guid)));
+                Assert.False(mc.CanConvertTo(context, typeof(object)));
+                Assert.False(mc.CanConvertTo(context, typeof(int)));
+            }
+        }
+
+        [Fact]
+        public void CreateInstance()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            MyTypeDescriptorContext context = new MyTypeDescriptorContext();
+
+            IDictionary values = new Dictionary<string, string>();
+            values.Add("Left", "1");
+            values.Add("Right", "2");
+            values.Add("Top", "3");
+            Assert.Throws<ArgumentException>(() => mc.CreateInstance(context, values));
+            values.Add("Bottom", "4");
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Throws<ArgumentException>(() => mc.CreateInstance(context, values));
+                return;
+            }
+            object result = mc.CreateInstance(context, values);
+            Assert.NotNull(result);
+
+            Assert.IsType<Margins>(result);
+            Margins margins = result as Margins;
+            Assert.Equal(1, margins.Left);
+            Assert.Equal(2, margins.Right);
+            Assert.Equal(3, margins.Top);
+            Assert.Equal(4, margins.Bottom);
+        }
+
+        [Fact]
+        public void GetCreateInstanceSupported()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            Assert.True(mc.GetCreateInstanceSupported(null));
+            Assert.True(mc.GetCreateInstanceSupported(new MyTypeDescriptorContext()));
+        }
+
+        [Fact]
+        public void ConvertFrom()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            CultureInfo culture = CultureInfo.InvariantCulture;
+
+            // try once with then once without context
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                object result;
+                Assert.Equal(',', culture.TextInfo.ListSeparator[0]);
+                if (PlatformDetection.IsFullFramework)
```

Comment:
What about the other ones like this one? 

The test string has semicolons for FullFramework and it has commas otherwise. Typo?

Reply:
Yes that has a different reason. Lines [73](https://github.com/mono/mono/blob/c5b88ec4f323f2bdb7c7d0a595ece28dae66579c/mcs/class/System.Drawing/System.Drawing.Printing/MarginsConverter.cs#L73) and [101](https://github.com/mono/mono/blob/c5b88ec4f323f2bdb7c7d0a595ece28dae66579c/mcs/class/System.Drawing/System.Drawing.Printing/MarginsConverter.cs#L101) in the mono code hardcode semicolon.

But from referencesource we see ConvertTo and ConvertFrom take the separator from `culture.TextInfo.ListSeparator`